pgrx/src/array.rs

```diff
@@ -373,10 +374,65 @@ impl RawArray {
         let ptr = self.ptr.as_ptr().cast::<u8>();
         ptr.wrapping_add(unsafe { varlena::varsize_any(ptr.cast()) })
     }
+
+    /// Slightly faster than new_array_type_with_len(0)
+    pub fn new_empty_array_type<T>() -> Result<RawArray, ArrayAllocError>
+    where
+        T: IntoDatum,
+        T: UnboxDatum,
+        T: Sized,
+    {
+        unsafe {
+            let array_type = pg_sys::construct_empty_array(T::type_oid());
+            let array_type =
+                NonNull::new(array_type).ok_or(ArrayAllocError::MemoryAllocationFailed)?;
+            Ok(RawArray::from_ptr(array_type))
+        }
+    }
+
+    /// Rustified version of new_intArrayType(int num) from https://github.com/postgres/postgres/blob/master/contrib/intarray/_int_tool.c#L219
+    pub fn new_array_type_with_len<T>(len: usize) -> Result<RawArray, ArrayAllocError>
+    where
+        T: IntoDatum,
+        T: UnboxDatum,
+        T: Sized,
+    {
```

Comment:
It's not okay to block something like:

```rust
let a = pgrx::datum::new_array_with_len::<&[u8]>(5).expect("failed to create array");
```

It's better to mark this function as unsafe, and avoid exposing anything other than those specific `Array::new_with_len` methods.

Reply:
Hum. Could you elaborate?